### PR TITLE
fwport: windows public clouds master

### PR DESCRIPTION
**Pull request by bogdanteleaga**
Forward port of windows public clouds on master.

Almost all of this is cherry-picked besides the last 3 commits which amount to:
- Removing some dead code
- Switching to the new os and series packages

### CODE REVIEW

**Comment by bogdanteleaga** [comment]
Added an extra commit with a couple of test fixes.

File: cloudconfig/powershell_helpers.go [hunk 1]
```diff
@@ -19,34 +28,34 @@ $ErrorActionPreference = "Stop"
 
 function ExecRetry($command, $maxRetryCount = 10, $retryInterval=2)
```
Comment (mattyw) [inline cloudconfig/powershell_helpers.go]: I'm afraid I really don't like doing this. If we need code in another file I'd much rather have that code in a seperate file and read it when needed. If we need a package var I'd read it in a package init()

http://paste.ubuntu.com/12530761/

Comment (bogdanteleaga) [inline cloudconfig/powershell_helpers.go]: Yup, I am aware this string is yucky, however it is outside the scope of this PR. We do have a trello card about it, should get solved when we get to it

Comment (mattyw) [inline cloudconfig/powershell_helpers.go]: @bogdanteleaga perfect, that's good enough for me

File: service/windows/zpassword_windows.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // go build mksyscall_windows.go && ./mksyscall_windows password_windows.go
 // MACHINE GENERATED BY THE COMMAND ABOVE; DO NOT EDIT
 
-package password
+package windows
 
 import "unsafe"
```
Comment (mattyw) [inline service/windows/zpassword_windows.go]: import (
"unsafe"
"syscall"
)

Why is unsafe being imported here?

**Comment by mattyw** [comment]
It would be nice to have some unit tests for the powershell code as well. Understand that's out of scope for this PR thought

**Comment by mattyw** [comment]
LGTM

**Comment by bogdanteleaga** [comment]
$$merge$$

**Comment by bogdanteleaga** [comment]
$$merge$$ rebased again to remove weird import that appeared because of the way the bot merges the branch

**Pull request merged**